File: neetcode/blind75/bit_manipulation/missing_number.py
# https://neetcode.io/problems/missing-number

class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        
        res = len(nums)

        for i in range(len(nums)):
            # if the correct number in the index it will result to 0 (e.g. 1 - 1)
            # if missing, it will result in -1 (e.g. 0 - 1), therefore shifting the value by -1
            res += i - nums[i]
        # after all the shifts by -1 when the value is mismatched, it will have the value that is missing
        return res

File: neetcode/neet150_JS/Bit-Manipulation/missing-number.js
// https://neetcode.io/problems/missing-number

class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    missingNumber(nums) {
        let n = nums.length
        let missing = n
        for (let i = 0; i < n; i ++) {
            missing ^= i ^ nums[i]
        }

        return missing
    }
}
